misc.c

```diff
@@ -440,7 +444,11 @@ GC_API void * GC_CALL GC_base(void * p)
         }
     if (HBLK_IS_FREE(candidate_hdr)) return(0);
     /* Make sure r points to the beginning of the object */
+#     if defined(__CHERI_PURE_CAPABILITY__)
+        r = cheri_align_up(r, WORDS_TO_BYTES(1));
+#     else
         r = (ptr_t)((word)r & ~(WORDS_TO_BYTES(1) - 1));
+#     endif
```

Comment:
Is `align_up` correct here? If we're trying to seek to the start of the object, shouldn't we `align_down`? I assume all objects are at least word-aligned, otherwise this alignment step doesn't make much sense anyway.

Reply:
Copy paste error - Thanks for catching that ! fixed in [fe2de5c](https://github.com/capablevms/bdwgc/pull/15/commits/fe2de5c1b33588e553821122f391a4d360ffbc7f)